From: Prarit Bhargava <prarit@redhat.com>
Subject: [PATCH]: AMD/ATI SB600 SMBus support
Date: Tue, 10 Apr 2007 07:57:46 -0400
Bugzilla: 232000
Message-Id: <20070410115746.7587.9908.sendpatchset@prarit.boston.redhat.com>
Changelog: [misc]: AMD/ATI SB600 SMBus support


Backport of

http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=4e6697fcc194db8b45559a9863947c6cbfeea363;hp=4ef0ce90df3763e277b5307fb580ff1cdaaad7b2

Successfully tested by me.

Resolves BZ 232000.

--- linux-2.6.18.x86_64.orig/Documentation/i2c/busses/i2c-piix4	2007-03-26 11:41:33.000000000 -0400
+++ linux-2.6.18.x86_64/Documentation/i2c/busses/i2c-piix4	2007-03-26 13:01:29.000000000 -0400
@@ -6,7 +6,7 @@ Supported adapters:
     Datasheet: Publicly available at the Intel website
   * ServerWorks OSB4, CSB5, CSB6 and HT-1000 southbridges
     Datasheet: Only available via NDA from ServerWorks
-  * ATI IXP southbridges IXP200, IXP300, IXP400
+  * ATI IXP200, IXP300, IXP400 and SB600 southbridges
     Datasheet: Not publicly available
   * Standard Microsystems (SMSC) SLC90E66 (Victory66) southbridge
     Datasheet: Publicly available at the SMSC website http://www.smsc.com
--- linux-2.6.18.x86_64.orig/drivers/i2c/busses/Kconfig	2007-03-26 11:41:32.000000000 -0400
+++ linux-2.6.18.x86_64/drivers/i2c/busses/Kconfig	2007-03-26 13:01:53.000000000 -0400
@@ -175,6 +175,7 @@ config I2C_PIIX4
 	    ATI IXP200
 	    ATI IXP300
 	    ATI IXP400
+	    ATI SB600
 	    Serverworks OSB4
 	    Serverworks CSB5
 	    Serverworks CSB6
--- linux-2.6.18.x86_64.orig/drivers/i2c/busses/i2c-piix4.c	2007-03-26 11:41:32.000000000 -0400
+++ linux-2.6.18.x86_64/drivers/i2c/busses/i2c-piix4.c	2007-03-26 13:05:16.000000000 -0400
@@ -23,6 +23,7 @@
    Supports:
 	Intel PIIX4, 440MX
 	Serverworks OSB4, CSB5, CSB6, HT-1000
+	ATI IXP200, IXP300, IXP400, SB600
 	SMSC Victory66
 
    Note: we assume there can only be one device, with one SMBus interface.
@@ -396,6 +397,8 @@ static struct pci_device_id piix4_ids[] 
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP400_SMBUS),
 	  .driver_data = 0 },
+	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP600_SMBUS),
+	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_OSB4),
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_CSB5),
--- linux-2.6.18.x86_64.orig/include/linux/pci_ids.h	2007-03-26 11:41:31.000000000 -0400
+++ linux-2.6.18.x86_64/include/linux/pci_ids.h	2007-03-26 13:05:45.000000000 -0400
@@ -362,6 +362,7 @@
 #define PCI_DEVICE_ID_ATI_IXP400_SATA2	0x437a
 #define PCI_DEVICE_ID_ATI_IXP600_SATA	0x4380
 #define PCI_DEVICE_ID_ATI_IXP600_SRAID	0x4381
+#define PCI_DEVICE_ID_ATI_IXP600_SMBUS	0x4385
 #define PCI_DEVICE_ID_ATI_IXP600_IDE	0x438c
 
 #define PCI_VENDOR_ID_VLSI		0x1004

